src/main/scala/uncore/tilelink2/Fuzzer.scala

```diff
@@ -34,16 +34,11 @@ class IDMapGenerator(numIds: Int) extends Module {
 
 object LFSR64
 { 
-  private var counter = 0 
-  private def next: Int = {
-    counter += 1
-    counter
-  }
-  
-  def apply(increment: Bool = Bool(true), seed: Int = next): UInt =
+  def apply(increment: Bool = Bool(true)): UInt =
   { 
     val wide = 64
-    val lfsr = RegInit(UInt((seed * 0xDEADBEEFCAFEBAB1L) >>> 1, width = wide))
+    val undef = Reg(UInt(width = wide)) // random value based on simulation seed
+    val lfsr = RegInit(Mux(undef === UInt(0), UInt(1), undef))
```

Comment:
I've inspected the verilog. It does do what was intended:

''''
  `ifdef RANDOMIZE_REG_INIT
  _GEN_1 = {2{$random}};
  _T_7 = _GEN_1[63:0];
  `endif


    if (reset) begin
      if (_T_9) begin
        lfsrs_0 <= 64'h1;
      end else begin
        lfsrs_0 <= _T_7;
      end
''''